tls/extensions/s2n_server_status_request.c

```diff
@@ -19,27 +19,54 @@
 #include "tls/s2n_tls_parameters.h"
 #include "tls/extensions/s2n_server_status_request.h"
 
-int s2n_server_extensions_status_request_send_size(struct s2n_connection *conn) {
-    if (s2n_server_can_send_ocsp(conn)) {
-        return 2 * sizeof(uint16_t);
-    }
+static bool s2n_server_status_request_should_send(struct s2n_connection *conn);
+static int s2n_server_status_request_send(struct s2n_connection *conn, struct s2n_stuffer *out);
+static int s2n_server_status_request_recv(struct s2n_connection *conn, struct s2n_stuffer *extension);
 
-    return 0;
+const s2n_extension_type s2n_server_status_request_extension = {
+    .iana_value = TLS_EXTENSION_STATUS_REQUEST,
+    .is_response = true,
```

Comment:
what does `is_response` mean here? if it's a requirement that a server requires the extension after a client sends it then it wouldn't be true.

Reply:
It's a requirement that the receiver/sender has already sent/received an extension with this iana_value that wasn't marked is_response. I know status_request can be requested by either side, which is why we track requests separately by sender: https://github.com/awslabs/s2n/blob/master/tls/s2n_connection.h#L98-L105

So here, if the s2n_server_status_request_extension is received by the server, the server must have sent a s2n_client_status_request_extension earlier in the handshake (yes, the naming is def off, I plan to fix that in a later PR).